### PR TITLE
ci: Add functionality to create defaults in docker compose

### LINKED ISSUE #2601: Add self-hosted docker environment variables for user login credentials to support automated testing
**Is your feature request related to a problem? Please describe.**

I am trying to incorporate flagsmith into my integration tests using testcontainers and I would like to be able to programmatically login so that I can obtain an App key to run various REST requests and SDK functions.

The docs suggest I need to manually create a user everytime i run a new docker container instance. I cant rely on volumes for persisting user data. https://docs.flagsmith.com/deployment/hosting/docker

I can see there are a variety of env vars supported: https://github.com/Flagsmith/self-hosted/blob/main/docker-compose.yml, would it be possible to add FLAGSMITH_USER, FLAGSMITH_PASSWORD in a similar fashion to postgres to give a predictable user to login with?

**Describe the solution you'd like**

Include FLAGSMITH_USER and FLAGSMITH_PASSWORD environment variables on the Flagsmith container to create a default user without having to manually register on startup.

**Describe alternatives you've considered**

Doing a postgres DB dump and pulling out the API key. This is obviously less than ideal.

**Additional context**

### PR DESCRIPTION
**Pull request by matthewelwell**
(updated by @khvn26)

## Changes

Adds `ALLOW_ADMIN_INITIATION_VIA_CLI` setting, set to `False` by default, that enables the `python manage.py bootstrap` management command.

The command detects an empty Flagsmith instance and initializes a superuser, an organisation, and a project. It does NOT do anything if the Flagsmith instance is not empty.

Additionally, the PR introduces individual commands for creating the default user, org, and project separately, with overridable defaults.

## How did you test this code?

Ran locally against a fresh database and one from a well-used Flagsmith instance, and observed expected behaviour.

### CODE REVIEW

File: api/core/management/commands/initselfhosted.py
Comment (khvn26) [inline api/core/management/commands/initselfhosted.py]: This and the following logs look like they should be command output via `self.stdout.write`.

File: docker-compose.yml
Comment (khvn26) [inline docker-compose.yml]: Init containers are cool but for the compose file I'd prefer too keep things as lean as possible to ease the cognitive load for the users.

Also `depends_on` isn't reliable without `condition` and throwing in `condition` makes us open a can of worms in regards to Compose's versioning.

**Review by khvn26 — DISMISSED** [review]

File: api/core/management/commands/initselfhosted.py
Comment (matthewelwell) [reply api/core/management/commands/initselfhosted.py]: Yep, makes sense, will change.

File: docker-compose.yml
Comment (matthewelwell) [reply docker-compose.yml]: Yeah, agreed. Will update.

File: docker-compose.yml
Comment (khvn26) [inline docker-compose.yml]: I don't at all love the idea of shipping commented code for the users to uncomment to enable something... just yuck.

Comment (dabeeeenster) [reply docker-compose.yml]: I cant really think of a neater way of doing this. It's also pretty common in docker-compose files to uncomment snippets to enable and disable services.

**Comment by laurencefass** [comment]
Im coming here from https://github.com/Flagsmith/flagsmith/issues/2601#issuecomment-1670893352. 

Comment says "fixed".

Is this now available in self hosted docker image?

Any idea when this will be available for deployment?

Thanks.

**Comment by matthewelwell** [comment]
> Im coming here from [#2601 (comment)](https://github.com/Flagsmith/flagsmith/issues/2601#issuecomment-1670893352).
> 
> Is this now available in self hosted docker image?
> 
> Any idea when this will be available for deployment?
> 
> Thanks.

Hi @laurencefass, this is not yet available anywhere. There is some additional work that needs to be done for this PR to be ready. We're hoping to finish it up in the next couple of weeks but other priorities have taken over a little. You can follow this PR and, once it's merged, you'll get a notification.

**Review by khvn26 — COMMENTED** [review]
Un-requesting changes, not approving as it's my code now that needs someone else's approval.

File: api/projects/management/commands/createinitialproject.py
Comment (khvn26) [inline api/projects/management/commands/createinitialproject.py]:
```suggestion
            logger.debug("Skipping initial project creation.")
```

File: api/organisations/management/commands/createinitialorganisation.py
Comment (zachaysan) [inline api/organisations/management/commands/createinitialorganisation.py]: It's strange to me that we allow no email and default to admin@example.com I take it this command is for people that want to spin something up really fast and use it right away?

File: api/tests/integration/ core/test_commands.py
Comment (zachaysan) [inline api/tests/integration/ core/test_commands.py]: This is cool.

File: api/users/forms.py
Comment (zachaysan) [inline api/users/forms.py]: Can `create_admin` be run multiple times? If so, the `create_initial_superuser` function name isn't quite right since it could be used for multiple superusers.

**Review by zachaysan — APPROVED** [review]
I have some questions, but you probably have good answers to them. Approved.

File: api/users/forms.py
Comment (khvn26) [reply api/users/forms.py]: > Can `create_admin` be run multiple times?

It's not supposed to be as the view using the form gets closed off after the user is created.

File: api/tests/integration/ core/test_commands.py
Comment (khvn26) [reply api/tests/integration/ core/test_commands.py]: I don't quite like monkeypatching (and ORM calls) in my integration tests... but thanks!

File: api/organisations/management/commands/createinitialorganisation.py
Comment (khvn26) [reply api/organisations/management/commands/createinitialorganisation.py]: Yes. The PR does add a `bootstrap` call to the entrypoint so the folks spinning up a new instance are going to get the superuser, org and project right away, unless `ALLOW_ADMIN_INITIATION_VIA_CLI` is set to `false`.

File: api/organisations/views.py
Comment (matthewelwell) [inline api/organisations/views.py]: This looks like a bad merge?

File: api/organisations/management/commands/createinitialorganisation.py
Comment (matthewelwell) [inline api/organisations/management/commands/createinitialorganisation.py]: What's the benefit to this approach vs setting the default in the parser argument above?

File: api/tests/integration/ core/test_commands.py
Comment (matthewelwell) [inline api/tests/integration/ core/test_commands.py]: Shouldn't these use the `expected_<entity>` arguments?

File: api/tests/integration/ core/test_commands.py
Comment (matthewelwell) [inline api/tests/integration/ core/test_commands.py:109]: These seem to be the same as above (which is why the string asserts above are passing still).

File: api/tests/integration/ core/test_commands.py
Comment (matthewelwell) [inline api/tests/integration/ core/test_commands.py]: I'm dubious as to whether these are valid tests. I don't quite know enough about the internals of the django ORM but I would be concerned that actually all we're doing here is e.g. `FFAdminUser.objects.all() == FFAdminUser.objects.all()` because there is no reason for django to evaluate the queryset until this point.

File: api/organisations/views.py
Comment (khvn26) [reply api/organisations/views.py]: Fixed! Thanks for catching this.

File: api/organisations/management/commands/createinitialorganisation.py
Comment (khvn26) [reply api/organisations/management/commands/createinitialorganisation.py]: Good point, updated.

File: api/tests/integration/ core/test_commands.py
Comment (khvn26) [reply api/tests/integration/ core/test_commands.py]: Agreed, done.

File: api/tests/integration/ core/test_commands.py
Comment (khvn26) [reply api/tests/integration/ core/test_commands.py:109]: Code above updated so this is resolved.

File: api/tests/integration/ core/test_commands.py
Comment (khvn26) [reply api/tests/integration/ core/test_commands.py]: This is a valid concern. Added the explicit list coercion.

File: api/organisations/management/commands/createinitialorganisation.py
Comment (matthewelwell) [reply api/organisations/management/commands/createinitialorganisation.py]: We can now change this to the following though, right? 

```python
organisation = Organisation.objects.create(name=organisation_name)
```

Comment (khvn26) [reply api/organisations/management/commands/createinitialorganisation.py]: Oops! Done.

**Review by matthewelwell — COMMENTED** [review]
As it's my original PR, I can't approve but ✅

**Pull request merged**
